### PR TITLE
added memory limit config

### PR DESCRIPTION
**Pull request by juliaElastic**
## What is the problem this PR solves?

Add memory limit config to set a soft limit: https://pkg.go.dev/runtime/debug#SetMemoryLimit

## How does this PR solve the problem?

This helps prevent OOM errors for quick agent actions on scale: https://weaviate.io/blog/gomemlimit-a-game-changer-for-high-memory-applications

## Design Checklist

- [ ] I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.
- [ ] I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.
- [ ] I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc.

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/fleet-server#changelog)

## Related issues

- Relates https://github.com/elastic/ingest-dev/issues/1674

### CODE REVIEW

File: changelog/fragments/1681906741-memory_limit.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#description:
+
+# Affected component; a word indicating the component this changeset affects.
+component:
+
+# PR URL; optional; the PR number that added the changeset.
+# If not present is automatically filled by the tooling finding the PR where this changelog fragment has been added.
+# NOTE: the tooling supports backports, so it's able to fill the original PR number instead of the backport PR number.
+# Please provide it if you are adding a fragment for a different PR.
+pr: https://github.com/elastic/fleet-server/pull/2514
```
Comment (michel-laterman) [inline changelog/fragments/1681906741-memory_limit.yaml:28]:
```suggestion
pr: 2514
```

File: internal/pkg/config/runtime.go [hunk 2]
```diff
@@ -5,7 +5,8 @@
 package config
 
 type Runtime struct {
-	GCPercent int `config:"gc_percent"`
+	GCPercent   int   `config:"gc_percent"`
+	MemoryLimit int64 `config:"memory_limit"`
```
Comment (michel-laterman) [inline internal/pkg/config/runtime.go:9]: Should we add a comment stating that this is set to `math.MaxInt64` by default; and setting this too low will result in a lot more GC behaviour (as outlined in the docs)?

**Comment by juliaElastic** [comment]
> Is the plan for this setting to be set alongside our other limits when we specify the max agents count? Will we be setting the soft limit as a part of #2308?

We discussed to set the limit in the cloud stackpack to be 50% of the container max memory: https://github.com/elastic/ingest-dev/issues/1674#issuecomment-1512811360

I started setting this in manual perf test runs to see what is the effect until we implement addition to the stackpack.

File: internal/pkg/config/runtime.go (hunk 2 above)
Comment (juliaElastic) [reply internal/pkg/config/runtime.go:11]: updated

**Pull request merged**